### ISSUE TITLE
Bar Charts show twice March instead of April

### ISSUE THREAD

**Issue by marcoruggine** [metabase/metabase#7286]
Hi guys, this is very unusual: since the new month, all my bar charts are showing twice "march" instead of "april". (I'm using metabase on docker with the  "JAVA_TIMEZONE=Europe/Rome" setting.)

I've spent days checking the data looking for a possible mistake in my queries but the dates are all correct, the same set of data shows correctly "april" if I switch to table view.

Any ideas on what is going on here?

![screen shot 2018-04-06 at 08 55 32](https://user-images.githubusercontent.com/20041428/38410505-098718d6-3985-11e8-8c21-1b91c387f569.png)
![screen shot 2018-04-06 at 08 55 20](https://user-images.githubusercontent.com/20041428/38410506-099f734a-3985-11e8-854f-b56e152b87e3.png)
<img width="1209" alt="schermata 2018-04-06 alle 10 22 38" src="https://user-images.githubusercontent.com/20041428/38410484-fad5f794-3984-11e8-85e8-fdfe5a385363.png">

**Comment by lucasloami** [comment]
@marcoruggine can you give more info? 

* What is your query result in form of table for each of the examples you displayed above?
* What is the data type of your "month column"?

I never faced a problem like this, but everytime I need to use "time field" in X axis, I usually use strings instead of data type "datetime". Maybe this option can help you as a quick'n'dirty approach until you figure out the real problem. =)

**Comment by marcoruggine** [comment]
Hi @lucasloami, I have checked my data too many times and I'm still confused... the result in table form is correct (see image attached)
<img width="632" alt="schermata 2018-04-11 alle 09 57 42" src="https://user-images.githubusercontent.com/20041428/38603794-e20cc5b0-3d6e-11e8-955f-3bcf33384256.png">

This is my query, as you can see it's very easy and to be safe I even casted the "time field". 

`SELECT 
    date::timestamptz as mese,
    area_manager,
    target_amount as obiettivo,
    round(sum(promo_amount - money_refund_amount), 0) as incasso_puc
FROM metabase_sales_targets_18
LEFT JOIN view_regional_history_new as v
    on metabase_sales_targets_18.region_name = 
        CASE WHEN v.region_name IN ('Lombardia','Lazio','Piemonte','Toscana','Veneto','Emilia-Romagna') 
                THEN v.region_name
            ELSE 'Altre Regioni' END
    AND date_trunc('month', metabase_sales_targets_18.date) = date_trunc('month',v.insert_date)
WHERE target_type = 'Reale'
    AND promo_amount IS NOT NULL
    AND area_manager ~* 'Stucchi'
    [[AND {{data}}]]
GROUP BY area_manager, date, target_amount
ORDER BY date ASC, incasso_puc DESC NULLS LAST`

If I cast it as a text, the results is different once again (image below). Everything is set back one month, but hovering over a column the hint shows the correct month.

<img width="1129" alt="schermata 2018-04-11 alle 10 02 40" src="https://user-images.githubusercontent.com/20041428/38604104-adfbc5d6-3d6f-11e8-8ad1-1939d3159211.png">

But if I use to_char(date, 'YYYY/MM') I have the correct chart (image below)

<img width="998" alt="schermata 2018-04-11 alle 10 03 47" src="https://user-images.githubusercontent.com/20041428/38604116-b40bb8be-3d6f-11e8-8e34-5c59391eb556.png">

So, to recap:

1. table format is correct
2. use to_char(date, 'YYYY/MM') is correct
3. native timestamptz shows March twice instead of April
4. casting as text brings everything back one month

I really don't know what to think!

I usually prefer to use the native timestamptz type because charts self adapt the amount of labels they show based on the dimension of the viewport. If I use a text or numeric type the labels will be packed and eventually will overlap each others.

**Comment by lucasloami** [comment]
Hum, interesting. I never faced this problem using Metabase. Probably because I don't use timestamptz in my questions. 

I read several times your comment above and everything seems to be correct, so my guess it should be some bug in Metabase. Maybe @mazameli can help us in this issue =)

**Comment by salsakran** [comment]
@marcoruggine what version of Metabase are you on? Can you please fill out the bug report template that was in the editor when you created this issue?

**Comment by marcoruggine** [comment]
- Your browser and the version: Chrome 65.0.3325.181
- Your operating system: macOS High Sierra 10.13.4
- Your databases: Postgres
- Metabase version: Metabase v0.28.1
- Metabase hosting environment: Docker

**Comment by lucasloami** [comment]
@marcoruggine I think a good start point for now it to update your Metabase to the latest stable version (v.0.28.6) and check if this problem was already solved and report here your results. 

If you need help to update your Metabase version, please check [this post](http://discourse.metabase.com/t/how-do-i-update-on-centos-7/3272/3) in Metabase forum.

**Comment by marcoruggine** [comment]
Hi @lucasloami , I've updated to 0.29 but unfortunately the anomaly still persists
<img width="1243" alt="schermata 2018-05-02 alle 09 54 14" src="https://user-images.githubusercontent.com/20041428/39511772-e71c6c8c-4dee-11e8-92b9-5caa18d8d6ee.png">

**Comment by jakobholmelund** [comment]
I can confirm this

version:     v0.29.0
database: Postgres

![screenshot_20180503_160815](https://user-images.githubusercontent.com/244930/39581632-4ef81966-4eec-11e8-9dde-43846f791a97.png)

It only happens when i use a different timezone than database default

**Comment by tkurki** [comment]
```SELECT x::date, 5
FROM   generate_series(timestamp '2018-01-01 00:00'
                     , timestamp '2018-05-02 00:00'
                     , interval  '1 month') t(x)```

- Metabase version  0.29.3
- psql timezone UTC, browser timezone EET

![image](https://user-images.githubusercontent.com/1049678/40363144-4ff48e22-5dd7-11e8-8581-c150fa02dbdb.png)

**Comment by tkurki** [comment]
If I do 
```
export TZ='Universal'
open /Applications/Google\ Chrome.app
```
I get just one March.

**Comment by Poggen** [comment]
We can confirm this as well.

Your browser and the version: Chrome 66.0.3359.139
Your operating system: macOS 10.13.4
Your databases: SQL Server
Metabase version: 29.3
Metabase hosting environment: Linux/Ubuntu 12
Metabase internal database: MySQL

![image 8](https://user-images.githubusercontent.com/5216834/40712110-2aced9fa-63fd-11e8-9ff1-825547b1e8c5.png)

**Comment by Linguk** [comment]
What is the progress on this bug? 

We use version 0.31.2 and this thing still exists.


Hope to see some fixes soon.

**Comment by mazameli** [comment]
Can anyone reproduce this on version 0.32 or 0.33?

**Comment by anastasiapapa** [comment]
Hey guys. We use version ```v0.33.0``` and this issue is still there for us, since quite some time now.

This is an example of a problematic card of how it looks in the dashboard it's part of, showing randomly only 2 months (not the correct months):

<img width="792" alt="Screenshot 2019-09-27 at 12 10 57" src="https://user-images.githubusercontent.com/43954136/65763086-db8b7c00-e122-11e9-909e-3b58ef1a069d.png">

And this is how it looks when I open it individually:

<img width="1652" alt="Screenshot 2019-09-27 at 12 25 56" src="https://user-images.githubusercontent.com/43954136/65763163-12fa2880-e123-11e9-95a6-d96f589cf763.png">

The only correct naming is from the first bar for March. 
If I hover over the bars then I see the correct month (like it shows "Date: August, 2019" in the above example). You can understand how confusing this can be for the end user.

FYI the data that is actually there, and should reflect on the naming, is from March to August. 

It would be nice to have a fix on this soon.
Cheers

**Comment by mazameli** [comment]
Just a quick update to report that I did a bit of naive testing of this on master, which has a bunch of timezone fixes in it. I did:

```
export TZ='IT'
open /Applications/Google\ Chrome.app 
```

 and set my report timezone in the Admin Panel to UTC, then to a couple other random different timezones. (It is highly likely I'm not testing this right. :b)

In each of these tests, I used @tkurki's example query on a postgres DB, but I couldn't make the double March bars appear: 

![image](https://user-images.githubusercontent.com/2223916/69386869-b9bcfa00-0c78-11ea-8224-3975df112e31.png)

**Comment by mazameli** [comment]
I have a fairly high suspicion this has been fixed since Nov 2019, and since we haven't received any additional comments about this bug since then, I'm going to go ahead and close this.

Related or possibly duplicate issues that we already closed as fixed:
#5246, #4245, #8413

**Comment by anastasiapapa** [comment]
Hello there @mazameli. Just to clarify what you are writing, that this is now fixed. I'm attaching the current state of the card I was showing in my comment above, from September.

<img width="1677" alt="Screenshot 2020-02-28 at 12 24 12" src="https://user-images.githubusercontent.com/43954136/75545347-605a2900-5a26-11ea-8bf6-15cf21d2b413.png">

As you can see, March was repeated twice, even though the seconds bars are carrying data for April as you see when I hover over them (exactly how it was before).
Now, what changed is that randomly since November, the month at the card is in agreement with the bar - actual date (October was skipped). So I'm guessing this is what you are referring too when you say that the issue is fixed starting November 2019. 
--> We shouldn't expect though that data before November would appear correctly, right? Just that from this point on this won't be an issue any more.

**Comment by flamber** [comment]
@anastasiapapa Could you post "Diagnostic Info" from Admin > Troubleshooting?

**Comment by joshi-hiren** [comment]
`{
  "browser-info": {
    "language": "en",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:88.0) Gecko/20100101 Firefox/88.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.11+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.11",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.11+9",
    "os.name": "Linux",
    "os.version": "4.14.231-173.360.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "Europe/London"
  },
  "metabase-info": {
    "databases": [
      "snowflake",
      "h2"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.5"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-04-27",
      "tag": "v0.39.1",
      "branch": "release-x.39.x",
      "hash": "6beba48"
    },
    "settings": {
      "report-timezone": "Europe/London"
    }
  }
}`

Would it be poosible to reopen this?
When clocks go forward (GMT->BST), we see an extra day and all of the days are off-set by one.
When clocks go back (BST->GMT), that day is skipped.

**Comment by joshi-hiren** [comment]
We have a table called CAL (just a load of dates in YYYY-MM-DD format).
`SELECT * FROM CAL ORDER BY cal_date` in the console:
```
...
2020-03-28
2020-03-29
2020-03-30
...
2020-10-24
2020-10-25
2020-10-26
...
```
But the same query on metabase:
```
...
28-3-2020, 24:00
29-3-2020, 24:00
29-3-2020, 24:00
30-3-2020, 24:00
...
24-10-2020, 24:00
26-10-2020, 24:00
...
```

The 29th March (the date we go into summer time) is repeated and the 25th October (the date we went into winter time is missing, this is consistant for every year with the missing/duplicate dates being the date when the clocks go forward/back.
